src/componententity.cpp

```diff
@@ -379,8 +373,6 @@ void ComponentEntity::ComponentEntityImpl::removeComponent(size_t index)
 {
     if (index < mComponents.size()) {
```

Comment:
Make sure that `index` is `>= 0`, and return `true`/`false`, as needed. (Same elsewhere in the code.)